pkg/providers/v1/aws_routes.go

```diff
@@ -112,9 +112,13 @@ func (c *Cloud) ListRoutes(ctx context.Context, clusterName string) ([]*cloudpro
 		// Capture instance routes
 		instanceID := aws.StringValue(r.InstanceId)
 		if instanceID != "" {
-			instance, found := instances[instanceID]
+			_, found := instances[instanceID]
```

Comment:
Do we need this check or can we remove the `instances, err := c.getInstancesByIDs(instanceIDs)` entirely?

Reply:
I don't know, I didn't think too much of it bc I think it's out of scope of this PR.

But I think the check is wrong  because ListRoutes should return all routes in the cluster's table and let the route controller decide whether it should be deleted. Because right now if there is a zombie route (say a node gets terminated while route controller is down) and then you hit the warning, the zombie will never get cleaned up?

but again I think it's out of scope so I'd rather fix this later

Reply:
> the zombie will never get cleaned up?

This is what I was wondering about, and I agree I'm pretty sure that check was only there due to necessity and we can remove it now.  The route controller can remove routes if the nodes don't exist: https://github.com/kubernetes/cloud-provider/blob/master/controllers/route/route_controller.go#L236-L257